services/basic_types.proto

```diff
@@ -1532,3 +1532,44 @@ message TokenAssociation {
     TokenID token_id = 1;  // The token involved in the association
     AccountID account_id = 2; // The account involved in the association
 }
+
+/**
+ * Staking metadata for an account or a contract returned in CryptoGetInfo or ContractGetInfo queries
+ */
+message StakingInfo {
+
+    /**
+     * If true, this account or contract declined to receive a staking reward.
+     */
+    bool decline_reward = 1;
+
+    /**
+     * The staking period during which either the staking settings for this account or contract changed (such as starting
+     * staking or changing staked_node_id) or the most recent reward was earned, whichever is later. If this account or contract
+     * is not currently staked to a node, then the value is -1.
+     */
+    int64 stake_period_start = 2;
+
+    /**
+     * The amount in tinybars that will be received in the next reward situation.
+     */
+    int64 pending_reward = 3;
+
+    /**
+     * The total of balance of all accounts staked to this account or contract.
+     */
+    int64 staked_to_me = 4;
+
+    /**
+     * The ID of the node this account or contract is staked to. If this account or contract is not staked to a node,
+     * the value is -1. The default value is -1.
+     */
+    int64 staked_node_id = 5;
+
+    /**
+     * The account to which this account or contract is staking. If not staked to an account, the value is null.
+     * The default value is null. It can never happen that staked_account_id is non-null and staked_node_id is
+     * non-negative at the same time.
+     */
+    AccountID staked_account_id = 6;
```

Comment:
make these into the same oneof staked_id as in Contract / Crypto create transaction body.

Reply:
Fixed. Thanks @xin-hedera